exporter/datasetexporter/datasetexporter.go

```diff
@@ -15,71 +15,107 @@
 package datasetexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datasetexporter"
 
 import (
-	"context"
 	"fmt"
-	"sync"
+	"net/http"
+	"reflect"
+	"strconv"
 	"time"
 
 	"github.com/google/uuid"
-	"go.opentelemetry.io/collector/pdata/plog"
-	"go.opentelemetry.io/collector/pdata/ptrace"
+	"github.com/scalyr/dataset-go/pkg/api/add_events"
+	"github.com/scalyr/dataset-go/pkg/client"
 	"go.uber.org/zap"
 	"golang.org/x/time/rate"
 )
 
-type datasetExporter struct {
-	limiter *rate.Limiter
-	logger  *zap.Logger
-	session string
+type DatasetExporter struct {
+	client      *client.DataSetClient
+	limiter     *rate.Limiter
+	logger      *zap.Logger
+	session     string
+	spanTracker spanTracker
 }
 
-var exporterInstance *datasetExporter
+func NewDatasetExporter(entity string, config *Config, logger *zap.Logger) (*DatasetExporter, error) {
```

Comment:
you shouldn't have to export this function, just use it in the factory.

Reply:
Thanks. I will review and fix all the other functions as well.